cime_config/buildlib

```diff
@@ -156,6 +156,57 @@ def _build_cam():
         _LOGGER.info("%s: \n\n output:\n %s \n\n err:\n\n%s\n", cmd, out, err)
         expect(retcode == 0, f"Command {cmd} failed with rc={retcode}")
 
+        # If dynamical core is MPAS, build it after CAM due to dependencies.
+        if dycore == "mpas":
+            _build_mpas(case)
+
+def _build_mpas(case: Case) -> None:
+    """
+    Build MPAS by invoking its build infrastructure.
+    """
+
+    case_root    = os.path.normpath(case.get_value("CASEROOT"))
+    atm_src_root = os.path.normpath(case.get_value("COMP_ROOT_DIR_ATM"))
+    atm_bld_root = os.path.normpath(os.path.join(case.get_value("EXEROOT"), "atm", "obj"))
+    lib_root     = os.path.normpath(os.path.join(case.get_value("EXEROOT"), "lib"))
+
+    gmake   = case.get_value("GMAKE")
+    gmake_j = case.get_value("GMAKE_J")
+
+    mpas_dycore_src_root = os.path.join(atm_src_root, "src", "dynamics", "mpas", "dycore", "src")
+    mpas_dycore_bld_root = os.path.join(atm_bld_root, "mpas")
+
+    # Make sure `mpas_dycore_src_root` exists. If not, it is likely that `./manage_externals/checkout_externals` did not succeed.
+    if os.path.isfile(mpas_dycore_src_root):
+        raise FileExistsError(1, "Unexpected file", mpas_dycore_src_root)
+
+    if not os.path.isdir(mpas_dycore_src_root):
+        raise FileNotFoundError(1, "No such directory", mpas_dycore_src_root)
+
+    # MPAS supports "in-source" build only. Copy source code to `mpas_dycore_bld_root` to avoid polluting `mpas_dycore_src_root`.
+    if os.path.isfile(mpas_dycore_bld_root):
+        raise FileExistsError(1, "Unexpected file", mpas_dycore_bld_root)
+
+    if not os.path.isdir(mpas_dycore_bld_root):
+        shutil.copytree(mpas_dycore_src_root, mpas_dycore_bld_root)
```

Comment:
How will this work if we modify an MPAS-A dycore source file in the CAM-SIMA source tree? If we subsequently run `case.build` in our case directory, will the modifications to the source be correctly incorporated?

Reply:
This should be addressed by https://github.com/ESCOMP/CAM-SIMA/pull/245/commits/0959d93b2baf5bd8ed98d9b77fd569c7df432059 below.